### PR TITLE
Cypress tests for unknown routes

### LINKED ISSUE #3505: Add local cypress test for unknown route error pages
**Is your feature request related to a problem? Please describe.**
Following https://github.com/bbc/simorgh/pull/2902 any unknown route will serve an error pages

**Describe the solution you'd like**
Hit the following routes, asserting their service, platform and errorcode is 404

/foobar => news canonical error page
/foobar.amp => news amp error page
/igbo/foobar => igbo canonical error page
/igbo/foobar.amp=> igbo amp error page

This cypress test should be local only, and probably will need to go in `cypress/integration/application/index.js`

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by PriyaKR**
Resolves #3505 

**Overall change:** _Added error page tests for unknown routes_

**Code changes:**

- _Visited few unknown error routes and asserted the page error according to the service if there is one available in the url or asserted for the default news one._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: cypress/integration/application/index.js [hunk 1]
```diff
@@ -39,3 +40,31 @@ describe('Application', () => {
     );
   });
 });
+
+describe('Application unknown route error pages', () => {
+  if (Cypress.env('APP_ENV') === 'local') {
+    var unknownRoutes = [
```
Comment (thekp) [inline cypress/integration/application/index.js]:
```suggestion
    const unknownRoutes = [
```

File: cypress/integration/application/index.js [hunk 2]
```diff
@@ -39,3 +40,31 @@ describe('Application', () => {
...
+      '/foobar',
+      '/foobar.amp',
+      '/igbo/foobar',
+      'igbo/foobar.amp',
+    ];
+    unknownRoutes.forEach(function(url) {
+      it('should display a news canonical error page', () => {
+        cy.visit(url, { failOnStatusCode: false });
+        let platform = 'news'
```
Comment (dr3) [inline cypress/integration/application/index.js]:
```
const service = url.includes('igbo') ? 'igbo' : 'news'
```

Bit cleaner imo, and `service` is more consistent across simorgh than `platform`

File: cypress/integration/application/index.js (hunk 1 above)
Comment (thekp) [reply cypress/integration/application/index.js]: we can ES6-ify some of the code below

**Review by dr3 — COMMENTED** [review]
Just wondering if theres anyway we can assert we get a 400, not a 500 status code. I know the test asserts the 404 status code visually in the page, but not the actual response code.

Cant think of a nice way to that though :/

File: cypress/integration/application/index.js (hunk 2 above)
Comment (PriyaKR) [reply cypress/integration/application/index.js]: yeah looks cleaner and less lines of code

Comment (PriyaKR) [reply cypress/integration/application/index.js]: Why do we want to assert for 400?

**Comment by dr3** [comment]
@PriyaKR Can we maybe add a `cy.testResponseCode` to double check its a 404 please?

File: cypress/integration/application/index.js (hunk 2 above)
Comment (dr3) [reply cypress/integration/application/index.js]: Incase we somehow start serving our 404 error page via a 500 status code. (its a 500, but it says 404 in the html)
Adding a `cy.testResponseCode` above the `visit` should do it :)

**Comment by PriyaKR** [comment]
> @PriyaKR Can we maybe add a `cy.testResponseCode` to double check its a 404 please?

Will look whether the response code be asserted using the same `cy.cisit` command if not have to add `testResponseCode`

File: cypress/integration/application/index.js (hunk 2 above)
Comment (PriyaKR) [reply cypress/integration/application/index.js]: Done

File: cypress/integration/application/index.js [hunk 3]
```diff
@@ -39,3 +40,29 @@ describe('Application', () => {
...
+
+describe('Application unknown route error pages', () => {
+  if (Cypress.env('APP_ENV') === 'local') {
+    const unknownRoutes = [
+      '/foobar',
+      '/foobar.amp',
+      '/igbo/foobar',
+      'igbo/foobar.amp',
+    ];
+    unknownRoutes.forEach(function(url) {
```
Comment (dr3) [inline cypress/integration/application/index.js]:
```suggestion
    unknownRoutes.forEach(url => {
```

**Review by dr3 — APPROVED** [review]

**Review by thekp — APPROVED** [review]
thanks for the updates 👍

**Pull request merged**